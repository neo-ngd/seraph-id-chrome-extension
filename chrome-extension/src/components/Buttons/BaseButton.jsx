// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';
import ArrowRight from '@material-ui/icons/KeyboardArrowRight';
import clsx from 'clsx';
import { useStyles } from './Styles';

/**
 * <BaseButton />
 * Primary button component.
 * @param handleClick
 * @param text
 * @param small
 * @param disabled
 * @param icon
 * @param fullWidth
 * @param reject
 * @param component
 * @param testID
 * @return {*}
 * @constructor
 */
const BaseButton = ({
  handleClick,
  text,
  small,
  disabled,
  icon,
  fullWidth,
  reject,
  component,
  testID,
}) => {
  const classes = useStyles();

  const style = {};
  if (disabled) {
    style.opacity = 0.5;
  }
  if (fullWidth) {
    style.width = '100%';
  }

  return (
    <Button
      data-test-id={testID}
      component={component}
      disabled={disabled}
      onClick={handleClick}
      variant="contained"
      color="inherit"
      className={clsx(
        classes.button,
        small && classes.smallButton,
        reject && classes.rejectButton
      )}
      style={style}
    >
      {text}
      {icon && (
        <ArrowRight
          style={{ opacity: disabled ? 0.5 : 1 }}
          className={clsx(classes.arrowRight, small && classes.smallArrowRight)}
        />
      )}
    </Button>
  );
};

BaseButton.defaultProps = {
  small: false,
  disabled: false,
  icon: false,
  fullWidth: false,
  reject: false,
  component: undefined,
  testID: 'base-button',
};

BaseButton.propTypes = {
  handleClick: PropTypes.func,
  text: PropTypes.string,
  small: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.bool,
  fullWidth: PropTypes.bool,
  reject: PropTypes.bool,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
};

export default BaseButton;
