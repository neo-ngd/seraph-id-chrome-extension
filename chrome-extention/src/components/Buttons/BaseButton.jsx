import React from 'react';
import NavigationIcon from '@material-ui/icons/Navigation';
import { Button } from '@material-ui/core';
import ArrowRight from '@material-ui/icons/KeyboardArrowRight';

import { useStyles } from './Styles';
import clsx from 'clsx';

export default function BaseButton({ handleClick, text, small = false, disabled = false, icon = false, fullWidth = true, reject = false }) {
  const classes = useStyles();

  const style = {};
  if (disabled) {
    style.opacity = .5;
  }
  if (fullWidth) {
    style.width = '100%'
  }

  return (
    <Button
      disabled={disabled}
      onClick={handleClick}
      variant="contained"
      color="inherit"
      className={clsx(classes.button, small && classes.smallButton, reject && classes.rejectButton)}
      style={style}
    >
      {text}
      {icon && (
        <ArrowRight className={clsx(classes.arrowRight, classes.smallArrowRight)} />
      )}
    </Button>
  );
}
