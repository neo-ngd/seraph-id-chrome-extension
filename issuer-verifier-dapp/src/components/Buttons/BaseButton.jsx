// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License
import React from "react";
import Button from "@material-ui/core/Button";

import { useStyles } from "./Styles";

export default function BaseButton({
  handleClick,
  text,
  disabled = false,
  icon = true
}) {
  const classes = useStyles();

  return (
    <Button
      disabled={disabled}
      onClick={handleClick}
      variant="contained"
      color="inherit"
      className={classes.button}
      style={disabled ? { opacity: 0.5 } : {}}
    >
      {text}
    </Button>
  );
}
