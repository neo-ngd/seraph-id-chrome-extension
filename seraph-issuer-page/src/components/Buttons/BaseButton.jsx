import React from "react";
import Button from "@material-ui/core/Button";
import { useStyles } from "./Styles";

export default function BaseButton({ handleClick, text }) {
  const classes = useStyles();
  return (
    <Button
      onClick={handleClick}
      variant="outlined"
      color="inherit"
      className={classes.button}
    >
      {text}
    </Button>
  );
}
