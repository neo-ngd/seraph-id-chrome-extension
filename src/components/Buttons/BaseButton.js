import React from "react";
import Button from "@material-ui/core/Button";
import { useStyles } from "./Styles";

export default function BaseButton({ handleClick, text }) {
  const classes = useStyles();
  return (
    <Button onClick={handleClick} className={classes.root}>
      {text}
    </Button>
  );
}
