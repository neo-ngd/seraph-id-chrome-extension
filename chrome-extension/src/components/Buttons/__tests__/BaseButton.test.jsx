import React from 'react';
import { shallow } from 'enzyme';
import BaseButton from "../BaseButton";

describe('<BaseButton />', () => {
    it('should render <BaseButton /> component', () => {
        const wrapper = shallow(<BaseButton />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render disabled component', () => {
        const wrapper = shallow(<BaseButton disabled={true}/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render component in full width', () => {
        const wrapper = shallow(<BaseButton fullWidth={true}/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render component with the icon', () => {
        const wrapper = shallow(<BaseButton icon={true}/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render as the small variant', () => {
        const wrapper = shallow(<BaseButton small={true}/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render as the "reject" variant', () => {
        const wrapper = shallow(<BaseButton reject={true}/>);
        expect(wrapper).toMatchSnapshot()
    });
});
