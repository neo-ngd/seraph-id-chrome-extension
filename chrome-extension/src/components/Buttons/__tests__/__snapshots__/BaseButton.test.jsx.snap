// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> should render <BaseButton /> component 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1"
  color="inherit"
  data-test-id="base-button"
  disabled={false}
  style={Object {}}
  variant="contained"
/>
`;

exports[`<BaseButton /> should render as the "reject" variant 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1 makeStyles-rejectButton-2"
  color="inherit"
  data-test-id="base-button"
  disabled={false}
  style={Object {}}
  variant="contained"
/>
`;

exports[`<BaseButton /> should render as the small variant 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1 makeStyles-smallButton-4"
  color="inherit"
  data-test-id="base-button"
  disabled={false}
  style={Object {}}
  variant="contained"
/>
`;

exports[`<BaseButton /> should render component in full width 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1"
  color="inherit"
  data-test-id="base-button"
  disabled={false}
  style={
    Object {
      "width": "100%",
    }
  }
  variant="contained"
/>
`;

exports[`<BaseButton /> should render component with the icon 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1"
  color="inherit"
  data-test-id="base-button"
  disabled={false}
  style={Object {}}
  variant="contained"
>
  <KeyboardArrowRightIcon
    className="makeStyles-arrowRight-3"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</WithStyles(ForwardRef(Button))>
`;

exports[`<BaseButton /> should render disabled component 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-button-1"
  color="inherit"
  data-test-id="base-button"
  disabled={true}
  style={
    Object {
      "opacity": 0.5,
    }
  }
  variant="contained"
/>
`;
